python/lsst/ts/externalscripts/base_take_twilight_flats.py

```diff
@@ -0,0 +1,457 @@
+                else:
+                    self.log.warning(
+                        f"Component {comp} not in CSC Group. "
+                        f"Must be one of {self.camera.components_attr}. "
+                        f"Ignoring."
+                    )
+
+        self.config = config
+
+        await super().configure(config)
+
+    def set_metadata(self, metadata: salobj.BaseMsgType) -> None:
+        """Set script metadata, including estimated duration."""
+
+        n_flats = self.config.n_flat
+
+        # Initialize estimate flat exposure time
+        target_flat_exptime = (self.config.max_exp_time + self.config.min_exp_time) / 2
+
+        # Setup time for the camera (readout and shutter time)
+        setup_time_per_image = self.camera.read_out_time + self.camera.shutter_time
+
+        # Total duration calculation
+        total_duration = (
+            self.instrument_setup_time  # Initial setup time for the instrument
+            + target_flat_exptime * (n_flats)  # Time for taking all flats
+            + setup_time_per_image * (n_flats)  # Setup time p/image
+        )
+
+        metadata.duration = total_duration
+        metadata.instrument = self.get_instrument_name()
+        metadata.filter = self.get_instrument_filter()
+
+    def get_new_exptime(self, sky_counts, exp_time):
+        """Calculate exposure time for next image.
+
+        Parameters
+        ----------
+        sky_counts : float
+            Counts in electrons of previous flat.
+        exp_time : float
+            Exposure time of previous flat
+
+        Returns
+        -------
+        float
+            Calculated new exposure time.
+        """
+
+        return self.config.target_sky_counts * exp_time / sky_counts
+
+    def get_target_radec(self):
+        """
+        Returns the RADEC of the target area of the sky that's an azimuth
+        `distance_from_sun` away from the Sun, given `elevation`,
+        and at a given `time`.
+
+        Parameters
+        ----------
+        distance_from_sun : float, (-180, 180)
+            The distance from the Sun in degrees.
+            Positive angles go towards the North.
+        elevation : float
+            Target elevation for Sky Flats.
+        """
+
+        az_sun, el_sun = self.tcs.get_sun_azel()
+
+        target_az = (az_sun + self.config.distance_from_sun) % 360
+
+        target_radec = self.tcs.radec_from_azel(target_az, self.config.target_el)
+
+        return target_radec
+
+    def get_empty_field(self, target, radius=5):
+        """
+        Query the "Deep blank field catalogue : J/MNRAS/427/679" in Vizier.
+
+        Parameters
+        ----------
+        target : astropy.coordinates.SkyCoord
+            Sky coordinates near the field
+        radius : float
+            Search radius in degrees.
+
+        Reference
+        ---------
+        http://cdsarc.u-strasbg.fr/viz-bin/Cat?J/MNRAS/427/679
+        """
+        _table = Vizier.query_region(
```

Comment:
You need to run this line here in an async executor, like I showed above for the consdb query. Should be something like this:

```
query_region = functools.partial(
    Vizier.query_region,
    catalog="J/MNRAS/427/679/blank_fld",
    coordinates=target,
    radius=radius * u.deg,
)

_table = await asyncio.get_event_loop().run_in_executor(
    None, query_region
)
```